src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,331 @@
+
+    // combine the block
+    if (po + block_length < signal_length) {       // current block is not the last block
+      signal->Range(po, block_length).CopyFromVec(signal_block_padded.Range(0, block_length));
+      signal->Range(po, filter_length - 1).AddVec(1.0, temp_pad);
+      temp_pad.CopyFromVec(signal_block_padded.Range(block_length, filter_length - 1));
+    } else {
+      signal->Range(po, signal_length - po).CopyFromVec(
+                        signal_block_padded.Range(0, signal_length - po));
+      if (filter_length - 1 < signal_length - po)
+        signal->Range(po, filter_length - 1).AddVec(1.0, temp_pad);
+      else
+        signal->Range(po, signal_length - po).AddVec(1.0, temp_pad.Range(0, signal_length - po));
+    }
+  }
+}
+
+void AddVectorsOfUnequalLength(const Vector<BaseFloat> &filter, Vector<BaseFloat> *signal) {
+  for (int po = 0; po < signal->Dim(); po += filter.Dim()) {
+    int block_length = filter.Dim();
+    if (signal->Dim() - po < block_length) block_length = signal->Dim() - po;
+    signal->Range(po, block_length).AddVec(1.0, filter.Range(0, block_length));
+  }
+}
+
+BaseFloat MaxAbsolute(const Vector<BaseFloat> &vector) {
+  float max = abs(vector.Max());
+  float max2 = abs(vector.Min());
+
+  if (max2 > max)
+    max = max2;
+
+  return max;
+}
+
+BaseFloat ComputeEarlyReverbEnergy(const Vector<BaseFloat> &rir, const Vector<BaseFloat> &signal,
+                                   BaseFloat samp_freq) {
+      float peak = 0;
+      int peak_index = 0;
+      peak = rir.Max(&peak_index);
+      KALDI_VLOG(1) << "peak value is " << peak;
+      KALDI_VLOG(1) << "peak_index is " << peak_index;
+
+      const float sec_before_peak = 0.001;
+      const float sec_after_peak = 0.05;
+      int early_rir_start_index = peak_index - sec_before_peak * samp_freq;
+      int early_rir_end_index = peak_index + sec_after_peak * samp_freq;
+      if (early_rir_start_index < 0) early_rir_start_index = 0;
+      if (early_rir_end_index > rir.Dim()) early_rir_end_index = rir.Dim();
+
+      int duration = early_rir_end_index - early_rir_start_index;
+      Vector<BaseFloat> early_rir(rir.Range(early_rir_start_index, duration));
+      Vector<BaseFloat> early_reverb(signal);
+      BlockConvolveSignals(early_rir, &early_reverb);
+
+      // compute the energy
+      return VecVec(early_reverb, early_reverb) / early_reverb.Dim();
+}
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+
+    const char *usage =
+        "Corrupts the wave files supplied via input pipe with the specified\n"
+        "room-impulse response (rir_matrix) and additive noise distortions\n"
+        "(specified by corresponding files).\n";
+
+    ParseOptions po(usage);
+    std::string noise_file;
+    BaseFloat snr_db = 20;
+    bool test_mode = false;
+    bool multi_channel_output = false;
+    int input_wave_channel = 0;
+    int rir_channel = 0;
+    int noise_channel = 0;
+
```

Comment:
Generally speaking, option explanations should begin with a capital letter.  There is no reason to say true|false, because the boilerplate text that is printed out explains that it's boolean and mentions the default value.